schemaless/create_uuid_map.py

```diff
@@ -17,25 +19,46 @@ def __init__(self):
     def from_files(cls, schemaless_file, uuid_map_file):
         """Build a RecordGraph from schemaless & uuid_map files."""
         rg = cls()
+
+        latest_records = latest_values(schemaless_file)
+
+        # Create a mapping between permit numbers and their associated PPTS
+        # record (so we an ensure they are assigned the same UUID).
+        permit_number_to_ppts = {}
+        for fk, record in latest_records[PPTS.NAME].items():
+            if record['building_permit_id']:
+                for permit_number in record['building_permit_id'].split(","):
+                    #  This assumes that a given permit can only have one PPTS
+                    #  parent. Is that accurate?
+                    permit_number_to_ppts[permit_number] = fk
+
         # Read the latest values from the schemaless file to build the graph.
-        for fk, record in latest_values(schemaless_file).items():
-            parents = []
-            children = []
-            if record['parents']:
-                parents = record['parents'].split(",")
-            if record['children']:
-                children = record['children'].split(",")
-            the_date = None
-            for source in source_map.values():
-                if source.DATE_KEY in record:
-                    the_date = source.get_date(record)
-            rg.add(Node(
-                record_id=fk,
-                date=the_date,
-                parents=parents,
-                children=children,
-                uuid=None,
-            ))
+        for source, source_records in latest_records.items():
+            for fk, record in source_records.items():
+                parents = []
+                children = []
+                if source == PPTS.NAME:
+                    if record['parents']:
+                        parents = record['parents'].split(",")
+                    if record['children']:
+                        children = record['children'].split(",")
+
+                if source == PTS.NAME:
+                    if permit_number_to_ppts[record['permit_number']]:
+                        parents = [permit_number_to_ppts[
+                            record['permit_number']]]
```

Comment:
Sorry stupid question since I'm trying to drive-by understand the code, how does this handle the permit number itself not being the fk, but the record id being the foreign key?

Reply:
If you look at the logic below where the Node actually gets added, things are keyed with the correct fk (which in the case of PTS will be record_id, as per the sources.py file). 

The permit_number_to_ppts dict is just something I computed above by iterating through PPTS records and looking at 'RELATED_BUILDING_PERMIT' for the purposes of intermediate computation -- it maps the list of PPTS fks that would be the "parent" for a given PTS record with a given permit number, since that is what people input in PPTS instead of the actual pts record id (tangential thought, right now some of the field naming 
for PPTS is a bit confusing -- if you look above, 'building_permit_id' is actually the field name that we mapped to 'RELATED_BUILDING_PERMIT', which is actually usually a permit number... I'll probably rename this in follow up).

Reply:
Ahh gotcha, thanks for the explanation, i think i misunderstood some of the field names but the code makes sense now